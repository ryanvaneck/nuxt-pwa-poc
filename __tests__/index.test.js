import { mount } from '@vue/test-utils'
import Index from '../pages/index.vue'

const options = {
  mocks: {
    $http: { get: () => {}}
  },
  stubs: ['NuxtLink'],
  data: function() {
    return {
      reservations: testData,
    }
  }
}

const testData = [
  {
    confirmationCode: 'AAAAAA',
    checkInDate: '2020-01-10',
    checkOutDate: '2020-02-14',
    city: 'San Francisco, CA',
    rating: null,
    cityImage:
      'https://cdn.pixabay.com/photo/2013/11/13/21/14/san-francisco-210230_960_720.jpg',
    locationDetail: 'Kasa Sand Diego at Aperture\nApartment 203'
  },
  {
    confirmationCode: 'BBBBBB',
    checkInDate: '2020-02-10',
    checkOutDate: '2020-03-14',
    city: 'Los Angeles, CA',
    rating: 3,
    cityImage:
      'https://cdn.pixabay.com/photo/2016/10/25/12/28/los-angeles-1768743_960_720.jpg',
    locationDetail: 'Kasa Sand Diego at Aperture\nApartment 203'
  },
  {
    confirmationCode: 'CCCCCC',
    checkInDate: '2020-04-10',
    checkOutDate: '2020-04-14',
    city: 'New York City, NY',
    rating: 5,
    cityImage:
      'https://cdn.pixabay.com/photo/2016/01/19/18/00/city-1150026_960_720.jpg',
    locationDetail: 'Kasa Sand Diego at Aperture\nApartment 203'
  }
];

describe('Index', () => {
  it('should match snapshot on first render', async () => {
    const wrapper = mount(Index, options)
    expect(wrapper.html()).toMatchSnapshot('first')
  });

  it('should display the reservations that match the search term', async () => {
    const wrapper = mount(Index, options)
    wrapper.find('.search-input').setValue('san')
    await wrapper.vm.$nextTick();

    const reservations = wrapper.findAll('.reservation');
    expect(reservations.length).toBe(1);
    const reservation = reservations.at(0);
    expect(reservation.html()).toMatchSnapshot('San Fran')
  });
})

