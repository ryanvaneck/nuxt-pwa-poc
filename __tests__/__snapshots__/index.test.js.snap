// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should display the reservations that match the search term: San Fran 1`] = `
"<div class=\\"reservation\\">
  <nuxtlink-stub to=\\"/AAAAAA\\">
    <div class=\\"city\\">San Francisco, CA</div>
    <div class=\\"dates\\">
      Jan 9 2020 - Feb 13 2020,
      <span> #AAAAAA</span></div>
  </nuxtlink-stub>
</div>"
`;

exports[`Index should match snapshot on first render: first 1`] = `
"<div>
  <div>header</div>
  <div class=\\"home\\">
    <div class=\\"hero\\"><img src=\\"~/assets/img/room.png\\" alt=\\"nice room\\" class=\\"hero-image\\">
      <div class=\\"search-box\\">
        <h1>Find Your reservation</h1>
        <h3>You can search by providing the city or the confirmation code of your reservation.</h3> <input type=\\"text\\" id=\\"searchTermId\\" class=\\"search-input\\"> <label for=\\"searchTermId\\"></label>
        <div class=\\"search-results\\"></div> <input type=\\"button\\" value=\\"Search\\" class=\\"search-button\\">
        <div></div>
      </div>
    </div>
  </div>
  <div class=\\"footer\\">footer</div>
</div>"
`;
